File: samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Domain/Models/CustomerInquiry.cs
// Copyright (c) Microsoft Corporation. All rights reserved.

namespace Microsoft.Teams.Samples.ConversationalTabs.Domain.Models;

using System.ComponentModel.DataAnnotations;

/// <summary>
/// Model for a CustomerInquiry, which is a SubEntity in our example.
/// </summary>
public record CustomerInquiry(
    string SubEntityId,
    DateTimeOffset CreatedDateTime,
    string CustomerName,
    string Question,
    string ConversationId,
    bool Active
): ISubEntity;

/// <summary>
/// Model to create a CustomerInquiry, passed in request body of the API
/// </summary>
public record CustomerInquiryInput(
    [MaxLength(9000)]
    string CustomerName,
    [MaxLength(9000)]
    string Question
);

File: samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Web/ClientApp/src/models/CustomerInquiry.ts
export type CustomerInquiry = {
  subEntityId: string;
  createdDateTime: string;
  customerName: string;
  question: string;
  conversationId: string;
  active: boolean;
};

export type CustomerInquiryInput = {
  customerName: string;
  question: string;
}
